backend/src/api/routes/moving_averages.py

```
@@ -0,0 +1,345 @@
+
+        # Extract and validate parameters
+        values = data.get('values')
+        ma_type = data.get('ma_type', 'sma')
+        window = data.get('window', 7)
+        weights = data.get('weights')
+
+        if not values or not isinstance(values, list):
+            raise HTTPException(
+                status_code=400,
+                detail="'values' must be a non-empty list of numbers"
+            )
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)